redisinsight/api/src/common/validators/cloud-job-data.validator.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  ValidationArguments,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+} from 'class-validator';
+import { isEmpty, isNumber } from 'lodash';
+import { CloudJobName } from 'src/modules/cloud/job/constants';
+import { CreateCloudJobDataDto, CreateCloudJobDto } from 'src/modules/cloud/job/dto';
+
+@ValidatorConstraint({ name: 'CloudJobDataValidator', async: true })
+export class CloudJobDataValidator implements ValidatorConstraintInterface {
```

Comment:
I don't think that it was good idea to create custom validator for possibly complex data structure
what if we add +1 field to the dto or +2. Seems like there is much more work to do manually. And even default message could not stay generic then.

Reply:
changed